### PR TITLE
Pass imagerefs to mco

### PR DESCRIPTION
**Pull request by csrwng**
**What this PR does / why we need it**:
In 4.12, the way the MCO is bootstrapped changed to take a single
image-references file instead of individual image flags. This commit
modifies the ignition-server to follow the same pattern when invoking
the MCO in bootstrap mode.

Includes the changes in #1764 so we can pass e2e

The MCO changes were introduced here: https://github.com/openshift/machine-config-operator/pull/3286
and corresponding installer change here: https://github.com/openshift/installer/pull/6234

**Checklist**
- [x] Subject and description added to both, commit and PR.

### CODE REVIEW

**Comment by cgwalters** [comment]
Looks great to me!  I am not sure how I somehow went off the rails on https://github.com/openshift/hypershift/pull/1768

File: ignition-server/cmd/run_local_ignitionprovider.go [hunk 1]
```diff
@@ -82,13 +83,25 @@ func (o *RunLocalIgnitionProviderOptions) Run(ctx context.Context) error {
...
+	}
+
 	p := &controllers.LocalIgnitionProvider{
 		Client:          cl,
 		ReleaseProvider: &releaseinfo.RegistryClientProvider{},
 		CloudProvider:   "",
 		Namespace:       o.Namespace,
 		WorkDir:         o.WorkDir,
 		PreserveOutput:  true,
+		ImageFileCache:  imageFileCache,
```
Comment (enxebre) [inline ignition-server/cmd/run_local_ignitionprovider.go:104]: I don't see where this is used

File: ignition-server/cmd/run_local_ignitionprovider.go [hunk 2]
```diff
@@ -3,6 +3,7 @@ package cmd
 import (
```
Comment (enxebre) [inline ignition-server/cmd/run_local_ignitionprovider.go:3]: any chance to reference the mco PR that force us to do this change?

File: ignition-server/cmd/run_local_ignitionprovider.go (hunk 1 above)
Comment (csrwng) [reply ignition-server/cmd/run_local_ignitionprovider.go:104]: This is for local debugging/testing

File: ignition-server/cmd/run_local_ignitionprovider.go (hunk 2 above)
Comment (csrwng) [reply ignition-server/cmd/run_local_ignitionprovider.go:3]: added to the PR desc

**Comment by csrwng** [comment]
Only Upgrade test failed in e2e-aws which is expected. Previous versions of the control plane operator will not come up without the changes in this PR.
/override ci/prow/e2e-aws

**Pull request merged**